lib/Client.php

```diff
@@ -134,7 +132,7 @@ public function transaction() {
      */
     public function close() {
         /** @var Promise $promise */
-        $promise = all(promises($this->promisors));
+        $promise = all($this->promisors);
```

Comment:
Why do we need two different implementations of Deferred, especially with a different interface?

If Deferred always will implement the Promise, such a function is generally not necessary.